inc/functions_upload.php

```diff
@@ -840,79 +840,104 @@ function add_attachments($pid, $forumpermissions, $attachwhere, $action=false)
 }
 
 /**
- * Delete an uploaded file both from the relative path and the CDN path if a CDN is in use.
+ * Delete an uploaded file both from the provided absolute path
+ * as well as from the associated CDN path if a CDN is in use.
  *
- * @param string $path The relative path to the uploaded file.
+ * @param string $abs_path The absolute path to the uploaded file.
+ *                         Must resolve to a directory under MYBB_ROOT.
  *
  * @return bool Whether the file was deleted successfully.
  */
-function delete_uploaded_file($path = '')
+function delete_uploaded_file($abs_path)
 {
 	global $mybb, $plugins;
 
 	$deleted = false;
 
-	$deleted = @unlink($path);
-
-	$cdn_base_path = rtrim($mybb->settings['cdnpath'], '/');
-	$path = ltrim($path, '/');
-	$cdn_path = realpath($cdn_base_path . '/' . $path);
-
-	if(!empty($mybb->settings['usecdn']) && !empty($cdn_base_path))
+	$realpath = realpath($abs_path);
+	if (substr($realpath, 0, strlen(MYBB_ROOT)) !== MYBB_ROOT || $realpath === realpath(MYBB_ROOT.'install/lock'))
 	{
-		$deleted = $deleted && @unlink($cdn_path);
+		return false;
 	}
+	else
+	{
+		$deleted = @unlink($realpath);
 
-	$hook_params = array(
-		'path' => &$path,
-		'deleted' => &$deleted,
-	);
+		$rel_path = substr($realpath, strlen(MYBB_ROOT));
+
+		if (!empty($mybb->settings['usecdn']))
+		{
+			$cdn_base_path = rtrim($mybb->settings['cdnpath'], '/');
+			if (!empty($cdn_base_path))
+			{
+				$cdn_path = realpath("{$cdn_base_path}/{$rel_path}");
+				$deleted = $deleted && @unlink($cdn_path);
```

Comment:
I think we should stop using the `@` operant like this because behaviour changed in php8+

Better to avoid in any new code or remove when updating specific lines, etc.

This may require a bit more of checks around the code.

Reply:
The suggestion is to avoid using the `@` operator and instead use `is_writable()` where applicable